AE-1601 Add virhetilastot.csv API

muep opened the pull request.

muep committed:
AE-1601 Add virhetilastot.csv API

muep commented on etp-backend/src/main/sql/solita/etp/db/valvonta-oikeellisuus.sql:
```diff
@@ -222,3 +222,20 @@ update vo_toimenpide set deleted = true where id = :toimenpide-id and publish_ti
+    extract(year from t.publish_time)::int as year,
+    extract(month from t.publish_time)::int as month,
+    v.type_id as type_id,
+    vt.label_fi as label_fi,
```
Olisi ehkä houkutus siirtää label viimeiseksi, niin olisi kaikki kapeat sarakkeet ensin ja tuolle epämääräisen pituiselle olisi sitten excelissä vapaasti tilaa

muep replied:
Siirsin labelin viimeiseksi, kun en oikeastaan keksinyt syytä sille että miksi sen pitäisi olla juuri tässä melkein viimeisenä

hkorpi commented on etp-backend/src/main/clj/solita/etp/api/valvonta_oikeellisuus.clj:
```diff
@@ -40,6 +40,15 @@
                                      uri (valvonta-service/add-virhetype!
                                            db (:body parameters))))}}]
+     ["/statistics/virhetilastot.csv"
+      {:conflicting true
```
onko tämä conflicting?

muep replied:
ei varmaan olekaan enää tällä polulla

muep committed:
AE-1601 Drop the :conflicting marker

muep committed:
AE-1601 Reorder columns so that label comes last

Having the long column last should make it more convenient to view
this in a spreadsheet tool

muep committed:
AE-1601 Use the last valvontamuistio of each ET

Previously, the code would look at all virhe objects of every
toimenpide.

muep commented on etp-backend/src/main/sql/solita/etp/db/valvonta-oikeellisuus.sql:
```diff
@@ -222,3 +222,28 @@ update vo_toimenpide set deleted = true where id = :toimenpide-id and publish_ti
+  from vo_toimenpide toimenpide
+  where
+    toimenpide.publish_time is not null and
+    toimenpide.type_id = 7 and
```
Pääsisiköhän tästä seiskasta eroon jotenkin kätevästi?

muep replied:
Todettiin että ilmeiset reitit ovat turhan työläitä.

hkorpi commented on etp-backend/src/main/clj/solita/etp/service/valvonta_oikeellisuus.clj:
```diff
@@ -74,6 +75,23 @@
+                                    (map (fn [k]
+                                           (k row))
+                                         col-keys))
+                                  virhetilastot))
```
(map (apply juxt col-keys) virhetilastot)

muep replied:
Vaihdoin yhden variaation tästä sinne tilalle.

hkorpi commented:
Tälle olisi hyvä tehdä yksi testi

muep committed:
AE-1601 Express CSV row formation through juxt

muep committed:
AE-1601 Document where type_id = 7 comes from

muep commented:
Merkkailin testin tarpeen tuonne AE-1578 kommentteihin.
